libevmasm/RuleList.h

```diff
@@ -374,6 +390,32 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart7(
 		false
 	});
 
+	rules.push_back({
+		// AND(A, SHR(B, X)) -> SHR(B, X) if 2^(256-B)-1 == A
+		{Instruction::AND, {A, {Instruction::SHR, {B, X}}}},
+		[=]() -> Pattern { return {Instruction::SHR, {B, X}}; },
+		false,
+		[=]() {
+			if (B.d() > std::numeric_limits<unsigned>::max())
```

Comment:
Ah right - boost multiprecision only has ``unsigned int`` shifts... actually then I'd even prefer ``B.d() >= 256`` here, sorry :-). We could in fact return true in this case (not that it matters too much, in practice this will never happen, since the rule above will catch this first I think).